code/modules/mining/equipment/monster_organs/monster_organ.dm

```diff
@@ -0,0 +1,202 @@
+/**
+ * Stabilising serum prevents monster organs from decaying before you can use them.
+ */
+/obj/item/mining_stabilizer
+	name = "stabilizing serum"
+	icon = 'icons/obj/medical/chemical.dmi'
+	icon_state = "bottle19"
+	desc = "Inject certain types of monster organs with this stabilizer to prevent their rapid decay."
+	w_class = WEIGHT_CLASS_TINY
+
+/obj/item/mining_stabilizer/afterattack(obj/item/organ/target_organ, mob/user, proximity)
+	. = ..()
+	if (!proximity)
+		return
+	var/obj/item/organ/internal/monster_core/target_core = target_organ
+	if (!istype(target_core, /obj/item/organ/internal/monster_core))
+		balloon_alert(user, "invalid target")
+		return
+
+	if (!target_core.preserve())
+		balloon_alert(user, "organ decayed")
+		return
+
+	to_chat(user, span_notice("You inject the [target_organ] with the stabilizer. It will no longer go inert."))
+	qdel(src)
+
+/**
+ * Useful organs which drop as loot from a mining creature.
+ * Generalised behaviour is that they will decay and become useless unless provided with serum.
+ * These should usually do something both when used in-hand, or when implanted into someone.
+ */
+/obj/item/organ/internal/monster_core
+	name = "monster core"
+	desc = "All that remains of a monster. This abstract item should not spawn. \
+		It will rapidly decay into uselessness. but don't worry because it's already useless."
+	icon = 'icons/obj/medical/organs/mining_organs.dmi'
+	icon_state = "roro core 2"
+	visual = FALSE
+	item_flags = NOBLUDGEON
+	slot = ORGAN_SLOT_MONSTER_CORE
+	organ_flags = NONE
+	force = 0
+	/// Set to true if this organ has decayed into uselessness.
+	var/inert = FALSE
+	/// ID of the timer which will decay this organ
+	var/decay_timer
+	/// Time after which organ should become useless
+	var/time_to_decay = 4 MINUTES
+	/// Icon state to apply when preserved
+	var/icon_state_preserved
+	/// Description to use once organ has been preserved
+	var/desc_preserved
+	/// Icon state to apply when inert
+	var/icon_state_inert
+	/// Description to use once organ is inert
+	var/desc_inert
+	/// Status effect applied by this organ
+	var/datum/status_effect/user_status
+	/// Moodlet applied by this organ
+	var/datum/mood_event/moodlet
+	/// Action to grant when organ is implanted
+	var/datum/action/item_action/organ_action/use_internal
```

Comment:
If you're going to add this much boilerplate, don't add it to a random subtype. Preservation should be an `/organ/internal` mechanic, as should `moodlet on insert` and `action to grant`